### PR TITLE
Improve CookieJar::addFromBrowserCookieCollection

### PR DESCRIPTION
**Pull request by szepeviktor**

### CODE REVIEW

File: src/Loader/Http/Cookies/CookieJar.php [hunk 1]
```diff
@@ -64,14 +64,18 @@ public function addFrom(string|UriInterface|Url $url, ResponseInterface|CookiesC
      */
     public function addFromBrowserCookieCollection(string|UriInterface|Url $url, CookiesCollection $collection): void
     {
-        if ($collection->count() > 0) {
-            $url = !$url instanceof Url ? Url::parse($url) : $url;
+        if ($collection->count() === 0) {
+            return;
+        }
```
Comment (szepeviktor) [inline src/Loader/Http/Cookies/CookieJar.php:69]: Return early.

File: src/Loader/Http/Cookies/CookieJar.php [hunk 2]
```diff
@@ -64,14 +64,18 @@ public function addFrom(string|UriInterface|Url $url, ResponseInterface|CookiesC
...
+
+        if (!$url instanceof Url) {
+            $url = Url::parse($url);
+        }
```
Comment (szepeviktor) [inline src/Loader/Http/Cookies/CookieJar.php:73]: Simplify `Url` instantiation.

File: src/Loader/Http/Cookies/CookieJar.php [hunk 3]
```diff
@@ -64,14 +64,18 @@ public function addFrom(string|UriInterface|Url $url, ResponseInterface|CookiesC
...
 
-            foreach ($collection as $cookie) {
-                $cookie = new Cookie($url, $this->buildSetCookieHeaderFromBrowserCookie($cookie));
+        foreach ($collection as $cookie) {
+            $setCookie = new Cookie($url, $this->buildSetCookieHeaderFromBrowserCookie($cookie));
```
Comment (szepeviktor) [inline src/Loader/Http/Cookies/CookieJar.php:76]: Do not reuse for's internal variable.

**Comment by otsch** [comment]
OK, why not.

**Pull request merged**